shared/utils/initCurrencyManager.ts

```diff
@@ -9,22 +9,75 @@ const defaultCurrencyIds = [
   "USDT-matic",
   "DAI-matic",
   "USDC-matic",
+  "AXS-mainnet",
+  "AUDIO-mainnet",
+  "RAI-mainnet",
+  "SYLO-mainnet",
+  "LDO-mainnet",
+  "UST-mainnet",
+  "MNT-mainnet",
+  "MIR-mainnet",
+  "INJ-mainnet",
+  "OCEAN-mainnet",
+  "ANKR-mainnet",
+  "RLY-mainnet",
+  "DAI-bsc",
+  "BUSD-bsc",
+  "USDC-xdai",
+  "USDC-avalanche",
+  "USDT-avalanche",
+  "USDC-optimism",
+  "USDT-optimism",
+  "DAI-optimism",
+  "USDC-multichain-moonbeam",
+  "USDC-wormhole-moonbeam",
+  "ETH-mainnet",
+  "REQ-mainnet",
+  "MATIC-matic",
+  "FTM-fantom",
+  "AVAX-avalanche",
+  "ETH-optimism-optimism",
+  "MNT-mantle",
+  "ETH-sepolia-sepolia",
+  "ETH-zksync-zksyncera",
+  "ETH-base-base",
+  "fUSDT-sepolia",
+  "fUSDC-sepolia",
 ];
 
+import { Types } from "@requestnetwork/request-client.js";
+
 export function initializeCurrencyManager(
   customCurrencies: any[]
 ): CurrencyManager {
   let currenciesToUse: any[];
 
-  if (customCurrencies?.length > 0) {
-    currenciesToUse = customCurrencies;
-  } else {
-    const defaultCurrencies = CurrencyManager.getDefaultList().filter(
-      (currency) => defaultCurrencyIds.includes(currency.id)
-    );
-    currenciesToUse = defaultCurrencies;
+  const defaultCurrencies = CurrencyManager.getDefaultList().filter(
+    (currency) => defaultCurrencyIds.includes(currency.id)
+  );
+
+  currenciesToUse = defaultCurrencies;
+
+  if (customCurrencies.length > 0) {
+    currenciesToUse.push(...customCurrencies);
   }
 
+  // Filter out duplicates based on a unique identifier
+  currenciesToUse = currenciesToUse.filter(
+    (currency, index, self) =>
+      index ===
+      self.findIndex((t) => {
+        if (currency.type === Types.RequestLogic.CURRENCY.ETH) {
+          return t.type === currency.type && t.network === currency.network;
+        } else if (currency.type === Types.RequestLogic.CURRENCY.ERC20) {
+          return (
+            t.network === currency.network &&
+            t.address?.toLowerCase() === currency.address?.toLowerCase()
+          );
+        }
+      })
+  );
```

Comment:
Why does this work?  I find this logic confusing. Is there a way to refactor it to be more clear?

`index === self.findIndex()` seems redundant and unnecessary, even if it isn't.

Reply:
Since now custom currencies are appended instead of overriding the default currencies there could be duplicates if we need to remove them.

The code itself checks on `network` and `address` in the case of ERC20 since those are the unique identifiers and for native currencies the main one is the `network` with the addition of `type`

`self.findIndex` would return to us the first element that meets the criteria,.

-  if it is the same index, we are comparing the same element which we should keep. 
- If it is not the same index it means we have a duplicate, and since the index of current element and the one found are not the same the equation returns false, removing the duplicate from the list.

@MantisClone